### PR TITLE
Drop Symfony 3.4 + Migrate Travis to github actions + Add php 8 support

### LINKED ISSUE #222: PHP 8 support
## Feature Request

It would be nice to add support to PHP 8 :)

Several questions to raise here:

* drop Symfony 3.4 support? last 3.4 bugfix release is next week IIRC
* replace Travis with Github Actions?
* what about phpstan? I see the CI allows failures for this run, but should it? maybe lower its level?

### PR DESCRIPTION
**Pull request by VincentLanglet**
Close https://github.com/FriendsOfSymfony/FOSCKEditorBundle/issues/222

### CODE REVIEW

**Comment by VincentLanglet** [comment]
I started working on https://github.com/FriendsOfSymfony/FOSCKEditorBundle/issues/222 @jmsche but the github actions is not triggered. Maybe you know why ?

Is there something to configure on the repository for github actions to work ? @lsmith77

**Comment by jmsche** [comment]
As it is not merged in this repository, actions are run in your own repo only, see https://github.com/VincentLanglet/FOSCKEditorBundle/actions

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Test
+
+on: [push, pull_request]
+
+jobs:
+    test:
+        name: PHP ${{ matrix.php-version }}
+
+        runs-on: ubuntu-18.04
```
Comment (jmsche) [inline .github/workflows/test.yaml:9]: Maybe run against ubuntu-20.04 instead?

**Comment by VincentLanglet** [comment]
> As it is not merged in this repository, actions are run in your own repo only, see https://github.com/VincentLanglet/FOSCKEditorBundle/actions

Thanks

**Review by jmsche — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (VincentLanglet) [reply .github/workflows/test.yaml:9]: 20.0.4 has only php 7.4+ pre-installed.

Comment (jmsche) [reply .github/workflows/test.yaml:9]: Not a problem as you don't use Ubuntu's PHP installed version, but shivammathur's

**Comment by jmsche** [comment]
You may want to change the PR's title as it also drop 3.4 & add PHP 8 support :)

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Install PHP with extensions
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  coverage: pcov
+                  tools: composer:v2
```
Comment (jmsche) [inline .github/workflows/test.yaml:34]:
```suggestion
                  tools: composer:v2
                  extensions: zip
```

**Review by jmsche — CHANGES_REQUESTED** [review]

**Comment by VincentLanglet** [comment]
@kunicmarko20 @lsmith77 This seems ready to be merged to me :)

Tests are green: https://github.com/VincentLanglet/FOSCKEditorBundle/actions/runs/424310014

**Comment by kunicmarko20** [comment]
Hey @VincentLanglet, sorry but I am not a maintainer of this package anymore, hopefully, @lsmith77 or @ElectricMaxxx could help with it.

**Comment by VincentLanglet** [comment]
> Hey @VincentLanglet, sorry but I am not a maintainer of this package anymore, hopefully, @lsmith77 or @ElectricMaxxx could help with it.

No problem, I had just trouble to find the maintainer of this package ; didn't find the information on the Readme.

File: .styleci.yml [hunk 3]
```diff
@@ -19,12 +19,7 @@ preset: symfony
 
 enabled:
   - combine_consecutive_unsets
-  - short_array_syntax
```
Comment (lsmith77) [inline .styleci.yml:22]: why did you make changes here?

**Comment by lsmith77** [comment]
honestly this package might not have an active maintainer right now.

but as always, I am willing to do the final review if people like you put int the effort.
in that spirit we can also discuss potentially you becoming the official maintainer.

I added a small question but otherwise it looks good indeed!

File: .styleci.yml (hunk 3 above)
Comment (VincentLanglet) [reply .styleci.yml:22]: I don't know but I got this error https://github.styleci.io/analyses/1beGJL?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Maybe the ci changed.

**Comment by VincentLanglet** [comment]
> honestly this package might not have an active maintainer right now.

It seems indeed.
 
> but as always, I am willing to do the final review if people like you put int the effort.

Thanks 

> in that spirit we can also discuss potentially you becoming the official maintainer.

I don't know much about this package, maybe with a "look for maintainer" issue, there will be better candidate.
I currently trying to add the php 8 compatibility on a multiple bundles. But I'm not against helping more.

**Comment by lsmith77** [comment]
thank you!

I guess only after merging it will be possible for you to confirm if the setup works properly.

**Comment by lsmith77** [comment]
so are you able to confirm it works now?
also I just noticed that we then also need to update the badges in the README to no longer point to travis-ci
https://github.com/FriendsOfSymfony/FOSCKEditorBundle/blob/2.x/README.md

**Comment by lsmith77** [comment]
see also #224

**Comment by VincentLanglet** [comment]
> so are you able to confirm it works now?

My project have other dependencies which are not compatible with PHP8, so I currently cannot test it.
I would recommend to create a release and see what happen.
- either it works for php 8
- either it doesn't, but it still works as good as currently ; people will submit issue that we can fix.

> also I just noticed that we then also need to update the badges in the README to no longer point to travis-ci
> https://github.com/FriendsOfSymfony/FOSCKEditorBundle/blob/2.x/README.md

Fixed in https://github.com/FriendsOfSymfony/FOSCKEditorBundle/pull/225

**Comment by jmsche** [comment]
@lsmith77 I just tested against a new sample Symfony app, using Symfony 5.2.1 & PHP 8.0.0 & it works :)

**Pull request merged**